### ISSUE TITLE
Account creation: failed payments should avoid creation new accounts !!

### ISSUE THREAD

**Issue by devforma22** [woothemes/woocommerce#30533]
Hello team ,

i have a critical problem ..(Bug)

In failed payement in paypal or other getways   ....automatecly the fonction of account creation is on ..there shouldn't be turn on ..
every one can see my private web site ...it's a big buug privacy of web sites..

Can you fix this issue plzz

maybe we should do some change in this part of code  "class -wc-checkout.php" ? (attach files)
![PB](https://user-images.githubusercontent.com/89191033/130046360-b24536b9-e632-408c-a453-9dafbebb64ef.PNG)

**Comment by tammullen** [comment]
Hi @devforma22 

Thank you for submitting the issue. However, you didn’t fill out the details of the bug report template that we ask for. Without these details, we can’t fully evaluate this issue. Please provide us with the information requested so we could take a look further.

**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

**Comment by devforma22** [comment]
**Describe the bug**
My site is a private course site ... you must pay for the course to have access to all the content on the site..

The bug is : the problem is when a payment is invalid or you exit the paypal gateway window ...... the user account is created this is a violent privatization of my site so, anyone can enter and access the content ......

**To Reproduce**

Steps to reproduce the behavior:

**Go to** 'page of order'  **next**  'complet all contente '  **next**  'clique in payement white paypal'  **after**  'you see the pop up of getway paypal'  **next**  'close  this pop up window  '  **after** refresh browser '  **you see the error or bug  the suprise :D** 'the customer account and already created without paid even without entering the bank information'  


**Screenshots**
1 : complet all input for order
![1](https://user-images.githubusercontent.com/89191033/130065108-ffb82dd9-a1c1-4853-8b51-7c117675010a.PNG)
2 : complet identification + password
![2](https://user-images.githubusercontent.com/89191033/130065136-939a8341-ac93-4d04-ade8-514b1f369b89.PNG)
3 : clique on order  paypal   then just close the pop up of payement (the bug is here)
![3](https://user-images.githubusercontent.com/89191033/130065164-829f2c20-b899-4258-87f4-4dbda9fc4ad7.PNG)
4 : after refrech browser the account is create  :(  all one can see the contente of my private web site 
![4](https://user-images.githubusercontent.com/89191033/130065186-6b0a6c4d-2645-45fb-9896-2d5a5b329c9b.PNG)
5: I think it is necessary to deactivate or put a condition on the trigger responsible for the creation of user accounts, but I cannot locate it
![PB](https://user-images.githubusercontent.com/89191033/130066653-af29ea6f-cc52-4220-b55d-b21f2a421a13.PNG)


**Expected behavior**

logically when a payment is not made or just we leave the payment window the account should not be created until a payment is validated 
-----I want the account to be created only after a valid payment---


**Isolating the problem (mark completed items with an [x]):**

 I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.  yes x
 This bug happens with a default WordPress theme active, or Storefront.  yes x
 I can reproduce this bug consistently using the steps above. yes x


**WordPress Environment**

Version de WooCommerce: 5.6.0
Environnement serveur
Info serveur :		Apache
Version PHP :		7.4.22
Poids Max Post PHP :		512 Mo
Limite d’exécution PHP :		90
PHP Max Input Vars :		1000
Version de cURL :		7.29.0, NSS/3.53.1
SUHOSIN installé :		–
Version MySQL :		5.5.5-10.4.20-MariaDB
Taille maximale de téléversement :		512 Mo

Base de données
Version de la base de données WooCommerce :		5.6.0
Préfixe de la base de données	 	wp_
Taille totale de la base de données	 	6.96MB
Taille de la base de données	 	5.63MB
Taille de l’index	 	1.33MB



Can you fix this issue plzz !!  @tammullen

**Comment by devforma22** [comment]
hello  team,

Can you help me solve the problem   plzzz

Tnk's

**Comment by devforma22** [comment]
Hello ,

is any one can helpe me to fix this bug pllz ?!!!  i'am blocked !! :(

Tnk's

**Comment by tammullen** [comment]
Hi @devforma22 

The number of details you provided is not enough for us to work with and identify what could be going on.

The steps provided are not clear enough to understand how to reproduce this.

We need full reproduction steps including details of the product types used and the settings you are using on WooCommerce > Settings > Accounts & Privacy

> after refrech browser the account is create :( all one can see the contente of my private web site

This does not look like a WooCommerce screenshot.

Finally, it will be great if you could share your full current System Status Report. You can get it by navigating to the WooCommerce / Status section of your site. Once there, click on the Get system report button and then copy it by clicking on the Copy for support button. Then paste it here in a comment.

**Comment by lkraav** [comment]
This problem is more than real, and a major design flaw in WooCommerce, that anybody who gets attacked with a botnet will soon realize.

Duplicate of https://github.com/woocommerce/woocommerce/issues/24603
